### PR TITLE
*: Fix HTTP api get TiFlash replica failed cause by concurrent DDL.

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve?

Fix https://github.com/pingcap/tidb/issues/17900

This Buf will affect TiFlash data replicate. 

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test.
- Manual test (add detailed scripts or steps below)

Side effects

- No

### Release note 

- 

- Fix HTTP api get TiFlash replica failed cause by concurrent DDL.

### CODE REVIEW

File: executor/ddl.go [hunk 1]
```diff
@@ -480,10 +480,14 @@ func GetDropOrTruncateTableInfoFromJobs(jobs []*model.Job, gcSafePoint uint64, d
...
-				fmt.Sprintf("(Schema ID %d)", job.SchemaID),
-				fmt.Sprintf("(Table ID %d)", job.TableID),
-			)
+			// The dropped/truncated DDL maybe execute failed that caused by the parallel DDL execution,
+			// then can't find the table from the snapshot info-schema. Such as below:
+			// time	|	session1				|	session2
+			// t1	|	drop table t1			|
+			// t2 	|	t1 state is write only	|	in other TiDB, and table t1 state is public, drop table t1
+			// t3 	|	finish					|
+			// t4 	|							|	finish with error: schema not exists.
```
Comment (wjhuang2016) [inline executor/ddl.go:489]: schema not exist?   or `table doesn't exist`?

**Comment by tangenta** [comment]
What is the output in step 4 and what is the expected output?

File: executor/ddl.go (hunk 1 above)
Comment (crazycs520) [inline executor/ddl.go:489]:
```suggestion
			// t4 	|							|	finish with error: table not exists.
```

File: executor/ddl.go [hunk 2]
```diff
@@ -480,10 +480,14 @@ func GetDropOrTruncateTableInfoFromJobs(jobs []*model.Job, gcSafePoint uint64, d
...
-				fmt.Sprintf("(Schema ID %d)", job.SchemaID),
-				fmt.Sprintf("(Table ID %d)", job.TableID),
-			)
+			// The dropped/truncated DDL maybe execute failed that caused by the parallel DDL execution,
+			// then can't find the table from the snapshot info-schema. Such as below:
+			// time	|	session1				|	session2
+			// t1	|	drop table t1			|
+			// t2 	|	t1 state is write only	|	in other TiDB, and table t1 state is public, drop table t1
+			// t3 	|	finish					|
+			// t4 	|							|	finish with error: table not exists.
```
Comment (crazycs520) [inline executor/ddl.go:489]:
```suggestion
			// t4 	|							|	finish with error: table doesn't exists.
```

**Comment by crazycs520** [comment]
@tangenta Already update the PR description, PTAL

File: executor/ddl.go [hunk 3]
```diff
@@ -480,10 +480,14 @@ func GetDropOrTruncateTableInfoFromJobs(jobs []*model.Job, gcSafePoint uint64, d
...
-				fmt.Sprintf("(Schema ID %d)", job.SchemaID),
-				fmt.Sprintf("(Table ID %d)", job.TableID),
-			)
+			// The dropped/truncated DDL maybe execute failed that caused by the parallel DDL execution,
+			// then can't find the table from the snapshot info-schema. Such as below:
+			// time	|	session1				|	session2
+			// t1	|	drop table t1			|
+			// t2 	|	t1 state is write only	|	in other TiDB, and table t1 state is public, drop table t1
+			// t3 	|	finish					|
+			// t4 	|							|	finish with error: table doesn't exists.
```
Comment (zimulala) [inline executor/ddl.go:489]: If this job is `history job`， do we need return an error？

**Pull request merged**